R/unit.conversions.R

```diff
@@ -343,6 +343,60 @@ flowconcToFluxConversion <- function(flow.units, conc.units, load.rate.units, at
   return(if(attach.units) multiplier else v(multiplier))
 }
 
+#' Get a conversion factor to convert between flow units
+#' 
+#' @importFrom unitted separate_units unitbundle get_units u v
+#' @export
+#' @param old.units character. The current units of flow.
+#' @param new.units character. The desired units of flow.
+#' @param attach.units logical. If TRUE, the conversion factor is returned with 
+#'   units attached.
+#' @return numeric conversion factor, to be multiplied by values in old units to
+#'   determine the values in new units.
+#' @examples 
+#' flowUnitsConversion(old.units='cfs', new.units='cms')
+#' flowUnitsConversion(old.units='m^3 s^-1', new.units='ft^3 s^-1')
+#' flowUnitsConversion(old.units='m^3 s^-1', new.units='ft^3 s^-1', attach.units=TRUE)
+#' 
+#' # use this multiplier to convert a vector
+#' Q_cfs <- seq(10, 12, length.out=10) # example data
+#' Q_cms <- Q_cfs * flowUnitsConversion(old.units='cfs', new.units='cms')
+flowUnitsConversion <- function(old.units, new.units, attach.units=FALSE) {
+  # Translate units - goes quickly if they're good already
+  old.units <- translateFreeformToUnitted(old.units, TRUE)
+  new.units <- translateFreeformToUnitted(new.units, TRUE)
+  
+  # Separate the units into numerator and denominator
+  old_separated <- separate_units(old.units)
+  old_numerstrs <- strsplit(get_units(unitbundle(old_separated[which(old_separated$Power > 0),])), " ")[[1]]
+  old_denomstrs <- strsplit(get_units(1/unitbundle(old_separated[which(old_separated$Power < 0),])), " ")[[1]]
+  new_separated <- separate_units(new.units)
+  new_numerstrs <- strsplit(get_units(unitbundle(new_separated[which(new_separated$Power > 0),])), " ")[[1]]
+  new_denomstrs <- strsplit(get_units(1/unitbundle(new_separated[which(new_separated$Power < 0),])), " ")[[1]]
+  
+  # Identify the volumetric pieces of the conversion from the units table
+  toL <- subset(unit.conversions, denominator == old_numerstrs & numerator == 'L')
+  fromL <- subset(unit.conversions, denominator == 'L' & numerator == new_numerstrs)
```

Comment:
is the `L` for liters? How does it know? I thought it might be from `get_units`, but I couldn't figure out where that function was defined in unitted.